File: net/OpenDraft.Lang/KeywordToken.cs
// <copyright file="KeywordToken.cs" company="Jon Rowlett">
// Copyright (c) Jon Rowlett. All rights reserved.
// </copyright>

namespace OpenDraft.Lang;

/// <summary>
/// Represents a keyword token in the language.
/// </summary>
public class KeywordToken : Token
{
    /// <summary>
    /// Initializes a new instance of the <see cref="KeywordToken"/> class.
    /// </summary>
    /// <param name="source">The source reference for this token.</param>
    /// <param name="value">The value of the keyword.</param>
    /// <exception cref="ArgumentNullException">Thrown if <paramref name="source"/> is null.</exception>
    /// <exception cref="ArgumentOutOfRangeException">Thrown if <paramref name="value"/> is <see cref="Keyword.None"/>.</exception>
    public KeywordToken(SourceReference source, Keyword value)
        : base(source)
    {
        this.Value = value != Keyword.None ?
            value :
            throw new ArgumentOutOfRangeException(nameof(value), "Value cannot be None.");
    }

    /// <summary>
    /// Gets the value of the keyword.
    /// </summary>
    public Keyword Value { get; }

    /// <inheritdoc/>
    public override string ToString()
    {
        return KeywordUtility.GetText(this.Value);
    }
}

File: java/src/io/github/localtoast9001/opendraft/lang/KeywordToken.java
/**
 * Copyright (c) Jon Rowlett. All rights reserved.
 */
package io.github.localtoast9001.opendraft.lang;

/**
 * Represents a keyword token in the OpenDraft language.
 */
public class KeywordToken extends Token {
    private final Keyword value;

    public KeywordToken(SourceReference source, Keyword value) {
        super(source);
        if (value == null) {
            throw new IllegalArgumentException("Value cannot be null");
        }
        this.value = value;
    }

    public Keyword getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "KeywordToken{" +
                "value=" + value +
                '}';
    }
}
